tests/parsers/nexcelom_matrix/nexcelom_matrix_parser_test.py

```diff
@@ -0,0 +1,28 @@
+import pytest
+
+from allotropy.parser_factory import Vendor
+from allotropy.testing.utils import from_file, validate_contents
+
+OUTPUT_FILES = ("nexcelom_matrix_1.xlsx",)
+
+VENDOR_TYPE = Vendor.NEXCELOM_MATRIX
+
+
+@pytest.mark.parametrize("output_file", OUTPUT_FILES)
+def test_parse_nexcelom_matrix_to_asm_expected_contents(output_file: str) -> None:
+    test_filepath = f"tests/parsers/nexcelom_matrix/testdata/{output_file}"
+    target_filename = output_file.replace(".xlsx", ".json").replace(".xls", ".json")
+    expected_filepath = f"tests/parsers/nexcelom_matrix/testdata/{target_filename}"
+    allotrope_dict = from_file(test_filepath, VENDOR_TYPE)
+
+    validate_contents(allotrope_dict, expected_filepath)
+
+
+# TODO This test fails with a KeyError instead of AllotropeConversionError
```

Comment:
The fix for this would be to wrap the accesses in a try/except, and raise `AllotropeConversionError` instead. This will be good to fix before merging.

To make this easier, you could wrap your `pd.Series` in the `SeriesData` class from `utils/pandas`, this lets you do type-safe accessors that specify if they are required or optional via a dict-like , and raises an `AllotropeConversionError` on failure!

e.g. instead of calling `value=sample["Viability"]`

`sample_data = SeriesData(sample)` <- this can be reused for all accesses
`value=sample_data[float, "Viability"]`

Reply:
Added in my most recent commit!